### PR TITLE
Update sync server

### PR DESCRIPTION
**Pull request by rorbech**
This updates to the latests sync server image (when an operational one is present) and migrates to use the _in-server image_ app import mechanism.

### CODE REVIEW

**Review by cmelchior — APPROVED** [review]
It is a bit annoying that we have to maintain two copies of apps, where previously we only had 1, but given how rarely we edit them, it probably isn't a huge burden. 

Getting rid of the hacky import script is a big 👍 

So if CI is happy (once we get it up and running again) 🚀

**Comment by rorbech** [comment]
The overall goal (besides getting something that could boot ;)) was to eliminate us having to maintain a separate app-import script. I guess we could have our `start_server` script to copy a template into the two `app_config` dirs. Only thing to change is the `name` in `config.json` so should be reasonable easy with a `jq`-replacement or similar.

**Comment by cmelchior** [comment]
> Only thing to change is the name in config.json so should be reasonable easy with a jq-replacement or similar.

Yes, I was thinking of something similar, so if that is relatively easy to do, I would probably prefer to do that. Rather than adding 20+ files that are just a copy of other files.

**Comment by rorbech** [comment]
Since the `app_config` dir is now also mounted to both the sync-server and the command server to communicate the app-ids  by the `app_config/<appname>/app_id` it would maybe also be worth to keep our initial template completely separate from the _running_ servers. So, having a template directory and cloning this into a separate runtime dir mounted into the images.

File: tools/sync_test_server/app_template/realm_config.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{ "name": "testapp1" }
```
Comment (rorbech) [inline tools/sync_test_server/app_template/realm_config.json:1]: Is this file required at all?

**Comment by clementetb** [comment]
Added templating for the app configurations as @cmelchior suggested.

It uses a temporary folder to store the actual templates to avoid polluting files. The actual app configurations are defined in the start_server script (line 42). Working on OSX.

File: tools/sync_test_server/start_server.sh [hunk 2]
```diff
@@ -34,13 +34,16 @@ adb reverse tcp:8888 tcp:8888 || { echo "Failed to reverse adb port." ; exit 1 ;
 # Make sure that Docker works correctly with Github Docker Registry by logging in
 docker login docker.pkg.github.com -u $GITHUB_DOCKER_USER -p $GITHUB_DOCKER_TOKEN
 
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+# Create app configurations
+APP_CONFIG_DIR=`mktemp -d`
+echo $APP_CONFIG_DIR
+$SCRIPTPATH/app_config_generator.sh $APP_CONFIG_DIR $SCRIPTPATH/app_template testapp1-hxsju testapp2-fbxtw
```
Comment (rorbech) [inline tools/sync_test_server/start_server.sh:42]: Any reason for having the actual ids in the naming. I think the pattern should be to place it into a folder with the app name (`testapp1` and `testapp2`) and then `run.sh` of the docker image will write the resulting app id into the `app_id` in the folder. If changing this you would also have to update the `mongodb-realm-command-server.js`. The app names should match those of https://github.com/realm/realm-java/blob/master/realm/realm-library/src/syncTestUtils/kotlin/io/realm/TestApp.kt#L31. 
```suggestion
$SCRIPTPATH/app_config_generator.sh $APP_CONFIG_DIR $SCRIPTPATH/app_template testapp1 testapp2
```

File: tools/sync_test_server/start_server.sh [hunk 3]
```diff
@@ -34,13 +34,16 @@ adb reverse tcp:8888 tcp:8888 || { echo "Failed to reverse adb port." ; exit 1 ;
 # Make sure that Docker works correctly with Github Docker Registry by logging in
 docker login docker.pkg.github.com -u $GITHUB_DOCKER_USER -p $GITHUB_DOCKER_TOKEN
 
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+# Create app configurations
+APP_CONFIG_DIR=`mktemp -d`
```
Comment (rorbech) [inline tools/sync_test_server/start_server.sh:40]: I guess this is a bit annoying if you would like to inspect the running setup, but I see the point of not running on top of previous garbage. Maybe just do a local temp dir so it is easier to find.

**Review by rorbech — COMMENTED** [review]
@clementetb There are a couple of oddities. Have you verified that out tests can run against this? The `mongodb-realm-command-server.js` should server the test app ids on `http::localhost:8888/<app-name>`, but I don't think that would work 🤔

**Comment by rorbech** [comment]
Something went wrong on my side because the `app_config` wasn’t empty, and it seems like it does quite some changes to the files, so I think maybe it is safest to use a real tmp-directory as you originally did. I guess if people really have to inspect it they could just go and modify the script temporarily…but now I put a final `echo` of the location, so it should be discoverable.
